prompb/write/v2/types.proto

```diff
@@ -18,47 +18,84 @@ option go_package = "writev2";
 
 import "gogoproto/gogo.proto";
 
-
+// WriteRequest represents a Remote Write 2.0 request to write given time series
+// to remote destination. Note that Remote Write 2.0 requires a content
+// negotiation for version, certain features and compressions, explained in
+// https://docs.google.com/document/d/1jx1fqpRnM0pAndeo3AgY7g6BLxN3Ah8R0Mm8RvNsHoU/edit
+// TODO(bwplotka): Change URL to Prometheus docs once ready.
 message WriteRequest {
+  // symbols contains de-duplicated array of string elements used for various
+  // items in WriteRequest like labels and some metadata items. To decode
+  // each of those items, referenced, by "ref(s)" suffix, you need to lookup the
+  // actual string by index from symbols array. The order of strings is up to
+  // the client, server should not assume any particular encoding.
   repeated string symbols = 1;
+  // timeseries represents multiple series, potentially from multiple targets.
   repeated TimeSeries timeseries = 2 [(gogoproto.nullable) = false];
 }
 
+// TimeSeries represents multiple series.
 message TimeSeries {
-  // Sorted list of label name-value pair references,
-  // encoded as indices to the strings array.
-  // This list's len is always multiple of 2.
-  repeated uint32 labels_refs = 1 ;
-
+  // labels_refs is a sorted list of label name-value pair references, encoded
+  // as indices to the WriteRequest.symbols array. This list's len is always
+  // a multiple of 2.
+  //
+  // Note that there might be multiple TimeSeries objects in the same
+  // WriteRequests with the same labels e.g. for different exemplars, metadata
+  // or created timestamp.
+  repeated uint32 labels_refs = 1;
+  // samples contain zero or more samples for a given timeseries. For a typical
+  // clients, in healthy cases, there will be only one sample, for ~real
+  // time metric streaming.
+  //
+  // Samples are sorted by time (older first).
+  // TODO(bwplotka): Can/should samples and histograms be in enum? Do we allow mix of those in one TimeSeries?
```

Comment:
If I understood your comment correctly, this discussion seems related: https://docs.google.com/document/d/1PljkX3YLLT-4f7MqrLt7XCVPG3IsjRREzYrUzBxCPV0/edit?disco=AAAA3dp46Gw

Reply:
Epic, let's discuss there + mtg, thanks!